### PR TITLE
MultiPointerTensor; better exception messages; other light improvements

### PR DESCRIPTION
**Pull request by iamtrask**
I also fixed a bug which prevented us from sending pointer -> pointer -> variables.

### CODE REVIEW

File: syft/exceptions.py [hunk 1]
```diff
@@ -42,3 +45,80 @@ class CannotRequestTensorAttribute(Exception):
     another tensor."""
 
     pass
+
+
+class TensorsNotCollocatedException(Exception):
```
Comment (robert-wagner) [inline syft/exceptions.py]: Please add a docstring explaining this error

File: syft/frameworks/torch/functions.py [hunk 2]
```diff
@@ -0,0 +1,5 @@
+from .tensors.interpreters import MultiPointerTensor
+
+
+def unite_pointers(*pointers):
```
Comment (robert-wagner) [inline syft/frameworks/torch/functions.py]: Combine pointers?

File: syft/workers/base.py [hunk 3]
```diff
@@ -333,7 +329,34 @@ def get_obj(self, obj_id):
...
+                    "Use .send() and .get() on all your tensors to make sure they're"
+                    "on the same machines.\n\n"
+                    "If you think this tensor does exist, check the ._objects dictionary"
+                    "on the worker and see for yourself!!! "
+                    "The most common reason this error happens is because someone calls"
+                    ".get() on the object's pointer without realizing it (which deletes "
+                    "the remote object and sends it to the pointer). Check your code to "
+                    "make sure you haven't already called .get() on this pointer!!!"
+                )
+                raise Exception(msg)
```
Comment (robert-wagner) [inline syft/workers/base.py]: Pick a specific type of exception

File: syft/frameworks/torch/functions.py (hunk 2 above)
Comment (iamtrask) [reply syft/frameworks/torch/functions.py]: are you recommending a name change or asking what teh functinoality is?

Comment (robert-wagner) [reply syft/frameworks/torch/functions.py]: Recommending a name change. I'm kinda either way on it though

Comment (iamtrask) [reply syft/frameworks/torch/functions.py]: i like combine - will change

File: syft/workers/base.py (hunk 3 above)
Comment (iamtrask) [reply syft/workers/base.py]: cool - will use KeyError

File: syft/workers/base.py [hunk 4]
```diff
@@ -333,7 +329,34 @@ def get_obj(self, obj_id):
...
+                    "Use .send() and .get() on all your tensors to make sure they're"
+                    "on the same machines.\n\n"
+                    "If you think this tensor does exist, check the ._objects dictionary"
+                    "on the worker and see for yourself!!! "
+                    "The most common reason this error happens is because someone calls"
+                    ".get() on the object's pointer without realizing it (which deletes "
+                    "the remote object and sends it to the pointer). Check your code to "
+                    "make sure you haven't already called .get() on this pointer!!!"
+                )
+                raise KeyErro(msg)
```
Comment (robert-wagner) [inline syft/workers/base.py]: KeyErro -> KeyError :slightly_smiling_face:

Comment (iamtrask) [reply syft/workers/base.py]: fixed

**Pull request merged**